CMakeLists.txt

```diff
@@ -1,12 +1,17 @@
 cmake_minimum_required(
-   VERSION 3.5
+   VERSION 3.9
```

Comment:
Starting from version `3.9`, `CMake` added additional support for `gtest`, making the output of `make test` a bit nicer. 
Currently, `3.10` is the CMake version in Ubuntu 18.04, so this version would still support the two most recent LTS releases. It is not a must-have version bump, so it could be worked around if we wanted to.